mail_tracking/models/mail_thread.py

```diff
@@ -40,3 +48,44 @@ def message_get_suggested_recipients(self):
                         record._message_add_suggested_recipient(
                             res, partner=partner, reason=_('Cc'))
         return res
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False,
+                        submenu=False):
+        """Add a filter to any model with mail.thread that will show up records
+           with tracking errors.
+        """
+        res = super().fields_view_get(
+            view_id=view_id, view_type=view_type, toolbar=toolbar,
+            submenu=submenu)
+        if view_type != 'search' and view_type != 'form':
+            return res
+        doc = etree.XML(res['arch'])
+        if view_type == 'search':
+            # Create filter element
+            new_filter = etree.Element(
+                'filter', {
+                    'string': _('Failed sent messages'),
+                    'name': "failed_message_ids",
+                    'domain': str([
+                        ('failed_message_ids.track_needs_action', '=', True)
+                    ])
+                })
+            # Modify view to add new filter element
+            nodes = doc.xpath("//search")
+            if nodes is not None and len(nodes) > 0:
+                nodes[0].append(etree.Element('separator'))
+                nodes[0].append(new_filter)
+        elif view_type == 'form':
+            # Create field
+            field_failed_messages = etree.Element('field', {
+                'name': 'failed_message_ids',
+                'widget': 'mail_failed_message',
+            })
+            # Modify view to add new filter element
+            nodes = doc.xpath(
+                "//field[@name='message_ids' and @widget='mail_thread']")
+            if nodes is not None and len(nodes) > 0:
```

Comment:
I guess this is the same:

```suggestion
            if nodes:
```